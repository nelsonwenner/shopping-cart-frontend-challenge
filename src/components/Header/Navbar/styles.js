import styled from 'styled-components'

export const Wrapper = styled.header`
  color: ${({ theme }) => theme.text.secondary};
  width: 100%;
  height: 100%;

  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;

  @media screen and (max-width: 1299px) {
    padding: 0 ${({ theme }) => theme.spacings.xsmall};
  }
`

export const Title = styled.h1`
  font-size: ${({ theme }) => theme.font.sizes.medium};
  font-weight: ${({ theme }) => theme.font.bold};
`

export const User = styled.div`
  font-size: ${({ theme }) => theme.font.sizes.xxsmall};
  font-weight: ${({ theme }) => theme.font.bold};

  display: flex;
  flex-direction: row;
  align-items: center;
  margin-left: auto;
  img {
    height: ${({ theme }) => theme.font.sizes.medium};
    width: ${({ theme }) => theme.font.sizes.medium};

    border-radius: 50%;
    margin-right: 1rem;
  }
`
