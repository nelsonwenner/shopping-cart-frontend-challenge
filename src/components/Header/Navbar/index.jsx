import React from 'react'

import * as S from './styles'

import Container from '@/components/common/Container/styles'
import avatar from '@/assets/avatar/default.png'

const Navbar = () => (
  <S.Wrapper as={Container}>
    <S.Title>Shopping</S.Title>
    <S.User>
      <img alt="avatar" src={avatar} />
      <span>John Doe</span>
    </S.User>
  </S.Wrapper>
)

export default Navbar
