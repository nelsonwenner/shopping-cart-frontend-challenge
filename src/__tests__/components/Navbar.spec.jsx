import React from 'react'
import { shallow } from 'enzyme'

import Navbar from '@/components/Header/Navbar'

describe('<Navbar>', () => {
  it('Should render Navbar component without fail and matches the snapshot', () => {
    const wrapper = shallow(<Navbar />)
    expect(wrapper).toMatchSnapshot()
  })
})
