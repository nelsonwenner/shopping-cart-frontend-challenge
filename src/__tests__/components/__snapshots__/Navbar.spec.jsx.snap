// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar> Should render Navbar component without fail and matches the snapshot 1`] = `
<styled.header
  as={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": e {
        "baseHash": 1948764880,
        "baseStyle": undefined,
        "componentId": "sc-hKwCoD",
        "isStatic": false,
        "rules": Array [
          "
  max-width: 1280px;
  width: 90%;
  margin-left: auto;
  margin-right: auto;

  @media only screen and (min-width: 993px) {
    .container {
      width: 70%;
    }
  }

  @media only screen and (min-width: 601px) {
    .container {
      width: 85%;
    }
  }
",
        ],
        "staticRulesId": "",
      },
      "foldedComponentIds": Array [],
      "render": [Function],
      "shouldForwardProp": undefined,
      "styledComponentId": "sc-hKwCoD",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
>
  <styled.h1>
    Shopping
  </styled.h1>
  <styled.div>
    <img
      alt="avatar"
      src="default.png"
    />
    <span>
      John Doe
    </span>
  </styled.div>
</styled.header>
`;
